crates/gen/src/types/enum.rs

```diff
@@ -33,14 +37,21 @@ impl Enum {
                     _ => panic!("Enum::from_type_def"),
                 };
 
+                let flags = field.flags(reader);
+                if underlying_type.is_none() && flags.is_static() && flags.literal() {
+                    let field_type = ElementType::from_blob(&mut field.sig(reader));
+                    underlying_type = Some(field_type);
+                }
+
                 fields.push((name, value));
             }
         }
 
         Self {
-            name,
-            fields,
-            signature,
+            name: name,
```

Comment:
`name: name` is redundant - why not just `name`?

Reply:
I honestly didn't know you could mix named and unnamed arguments like that, so I learned something today!